File: sdkwork-ai-sdk-go/types/music_generation_request.go
package types

// Music generation request
type MusicGenerationRequest struct {
	Model string `json:"model"`
	Prompt string `json:"prompt"`
	Duration int `json:"duration"`
	Format string `json:"format"`
	N int `json:"n"`
	Style string `json:"style"`
	Metadata map[string]string `json:"metadata"`
	NegativePrompt string `json:"negative_prompt"`
	ResponseFormat string `json:"response_format"`
	ReferenceAudio string `json:"reference_audio"`
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/models/music_generation_request.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class MusicGenerationRequest:
    """Music generation request"""
    model: str
    prompt: str
    negative_prompt: str = None
    duration: int = None
    format: str = None
    n: int = None
    reference_audio: str = None
    style: str = None
    metadata: Dict[str, str] = None
    response_format: str = None
